pkg/crc/api/api.go

```diff
@@ -0,0 +1,91 @@
+package api
+
+import (
+	"bytes"
+	"encoding/json"
+	"fmt"
+	"net"
+
+	"github.com/code-ready/crc/pkg/crc/logging"
+)
+
+type ArgsType map[string]string
+type handlerFunc func(ArgsType) string
+
+type CrcApiServer struct {
+	listener net.Listener
+	handlers map[string]handlerFunc // relates commands to handler func
+}
+
+// request struct is used to decode the json request from tray
+type request struct {
```

Comment:
even jsonRequest is too explicit about the type and possible conflict. You should try to avoid using the variable type or implementation out of the naming, so something like `commandRequest` or `commandReceived` would be more appropriate.